### ISSUE TITLE
Toggle console with a single key

### ISSUE THREAD

**Issue by Obegg** [mpv-player/mpv#10614]
Currently - to open console you have to press 2 keys, its not a toggle. Is there any way to make it show and hide console by only one key?

**Comment by christoph-heinrich** [comment]
``` lua
mp.add_forced_key_binding("esc", nil, function()
	mp.commandv("script-binding", "console/enable")
end)
```
With that you can open and close the console with escape.

**Comment by Hrxn** [comment]
^ and this wouldn't work in `input.conf`?

**Comment by christoph-heinrich** [comment]
> ^ and this wouldn't work in `input.conf`?

No, I tried it in input.conf first, but apparently it has to be a forced key bind to overwrite the escape key.

**Comment by Hrxn** [comment]
But other keys work as expected here for the console toggle?

**Comment by christoph-heinrich** [comment]
The console uses escape for closing it, and you can't change that without changing the code. And so if the goal is toggling, and closing has to be escape, then opening also has to be escape.

**Comment by Obegg** [comment]
So complicated, for what? the stats (frame dropped and etc) is a toggle, why cant console be the same?

**Comment by Obegg** [comment]
> ```lua
> mp.add_forced_key_binding("esc", nil, function()
> 	mp.commandv("script-binding", "console/enable")
> end)
> ```
> 
> With that you can open and close the console with escape.
> 
> Copy that text into a `.lua` file in your [scripts folder](https://mpv.io/manual/stable/#script-location).

The thing is - I don't want to use escape key as the toggle, let's say I want to use `g` (example), it doesn't work.

**Comment by christoph-heinrich** [comment]
Alternatively you make a PR for add a binding for disabling to the console, analogous to it's [enable](https://github.com/mpv-player/mpv/blob/d433c5d655b79c957dde329b7fa6f58e7f16021d/player/lua/console.lua#L781-L783) binding. Then you can use any any key binding for toggling that does not get overwritten when console is open. You can find the key bindings that get enabled upon opening the console [here](https://github.com/mpv-player/mpv/blob/d433c5d655b79c957dde329b7fa6f58e7f16021d/player/lua/console.lua#L689-L738).

**Comment by sfan5** [comment]
Not sure what the commotion is about, you can easily bind it to the key you want. Here's what I use:
<pre>
F9 script-binding console/enable
</pre>

**Comment by guidocella** [comment]
This was discussed in https://github.com/mpv-player/mpv/pull/10075#issuecomment-1097343116.

I thought that if the console enabled and disabled an input section instead of binding and unbinding the keybindings on open and close, you could add bindings like `TAB {console} keypress ESC`. But I later noticed that rossy's original script at https://github.com/rossy/mpv-repl/blob/master/repl.lua did use input sections, and wm4 removed them when merging it, which must be because he deprecated input sections, though with him gone most likely no one is going to remove them. So I'm hesitant about restoring them. And even with an input section you could only invoke other keypresses and not console.lua's functions directly, though they could be converted to script messages.

He actually forgot to remove the line `mp.enable_key_bindings('console-input', 'allow-hide-cursor+allow-vo-dragging')` so you can add `{console-input}` keybindings, as long as they're not the same as other console.lua keybindings since it uses `add_forced_key_binding`, but they are never disabled, e.g.

```
F1 script-binding console/enable
F1 {console-input} keypress esc
```

Pressing F1 twice toggles the console but pressing it again no longer enables it.

**Comment by garoto** [comment]
I raised this question on IRC just after rossy's script was adapted and made into a script embedded into the final mpv binary. wm4 said it was reasonable for the same to key to open the console to be the same key to close it. But I guess he was just not interested in messing with this at that point.

**Comment by eugenesvk** [comment]
> because he deprecated input sections

what is the supposed replacement to these?

**Comment by guidocella** [comment]
There is no replacement, you should keep using input sections since the developer that wanted to remove them left.

**Comment by guidocella** [comment]
You can just use this script actually:

```lua
local open = false

mp.add_key_binding('F1', function ()
    mp.commandv('script-message-to', 'console', open and 'disable' or 'enable')
    open = not open
end)
```

**Comment by Hrxn** [comment]
@guidocella Does this script example also work with
```
[input] Key ` is bound to:
[input] 1.  'script-binding console/enable' in <builtin>:1 (default)
```

(i.e. the standard console key), or does this break something else?

**Comment by guidocella** [comment]
Nope it only works with keys not bound by console.lua.

Scripts go in .lua files in in the `scripts` directory, not in `input.conf`.

**Comment by guidocella** [comment]
Just give a name to the binding as the second argument

```lua
local open = false

mp.add_key_binding(nil, 'toggle-console', function ()
    mp.commandv('script-message-to', 'console', open and 'disable' or 'enable')
    open = not open
end)
```

and bind `script-message toggle-console`

**Comment by jimmy-1000** [comment]
> Finally I think I can close this issue, thank you!

Hi. Not very clear for me. Can you now open and close the console with the same key (other than the escape key)?

**Comment by guidocella** [comment]
It is even simpler now, just use this script replacing F1 with the desired key

```lua
mp.add_key_binding('F1', 'toggle-console', function ()
    mp.command(mp.get_property_native('user-data/mpv/console/open') and 'keypress ESC' or 'script-binding commands/open')
end)
```

**Comment by jimmy-1000** [comment]
> It is even simpler now, just use this script replacing F1 with the desired key
> 
> mp.add_key_binding('F1', 'toggle-console', function ()
>     mp.command(mp.get_property_native('user-data/mpv/console/open') and 'keypress ESC' or 'script-binding commands/open')
> end)

Thanks Guidocella 👌
